packages/build/src/plugins_core/edge_functions/validate_manifest/validate_edge_functions_manifest.js

```diff
@@ -0,0 +1,100 @@
+import { promises as fs } from 'fs'
+import { createRequire } from 'module'
+import { join, resolve } from 'path'
+
+import Ajv from 'ajv'
+
+import { addErrorInfo } from '../../../error/info.js'
+import { log } from '../../../log/logger.js'
+import { hasEdgeFunctionsDirectories } from '../utils.js'
+
+const require = createRequire(import.meta.url)
+const ajv = new Ajv({ allErrors: true })
+require('ajv-errors')(ajv)
+
+// regex pattern for manifest route pattern
+// checks if the pattern string starts with ^ and ends with $
+// we define this format in edge-bundler:
+// https://github.com/netlify/edge-bundler/blob/main/src/manifest.ts#L66
+const normalizedPatternRegex = /^\^.*\$$/
```

Comment:
Our regular expressions need to be compatible with Go, which means not supporting certain features that JS regexes support (like negative lookahead). I imagine that it'll be difficult to check for that with our validation rules, but if we can find a way, that would be fantastic!